Dockerfile

```diff
@@ -0,0 +1,15 @@
+FROM node:latest
+
+WORKDIR /app
+
+COPY package*.json .
+
+RUN npm ci
+
+COPY . .
+
+RUN npm run build
+
+EXPOSE 5173
+
+ENTRYPOINT npm run preview
```

Comment:
any reason to take this over the CMD? Are there any command suffixes it'd make sense to add here?

Reply:
Just to add here: unless you figured out some special vite sauce I don't know, you'll probably have to add the bulls step as part of the entry point / cmd to ensure the runtime env variables get used here

Reply:
> any reason to take this over the CMD? Are there any command suffixes it'd make sense to add here?

I reckon we didn't want it to be overwritten.


> you'll probably have to add the bulls step as part of the entry point / cmd to ensure the runtime env variables get used here

build step? It's a couple of lines above it.

Reply:
Right, but this build step runs only once during image creation the way you set it up. I assume at that point you've made a bunch of static files that then get served with `npm run preview` whenever I actually launch a container from this image. So I'm wondering how you ensure that I can do this:

- docker build without providing the ENV variables (i.e. including not in my shell while building) and
- changing the ENV var every time I launch a new container from the image

In the old tool we had to build the app every time someone launched it: https://github.com/neurobagel/old-query-tool/blob/10ba7fce2b8903338386f466b81d63bcd24ee82e/Dockerfile#L15
So I'm wondering what you are doing here that makes this unnecessary

Reply:
to be clear: what we did with the old app wasn't great or smart. It was just a simple way around the problem of having the ENV vars baked into the static app. There are ways around this that don't require rebuilding, but afaik they involve some form of search-replace within the built app dist folder.

Reply:
> I reckon we didn't want it to be overwritten.

That's `ENTRYPOINT`. `CMD` can be overwritten: https://docs.docker.com/reference/dockerfile/#entrypoint . But you can append commands to `ENTRYPOINT`

Reply:
aha aha, my bad. You and I are saying the same thing. All good on ENTRYPOINT then